### ISSUE TITLE
When the heading level skips a number or two, toc cannot recongnize them. 

### ISSUE THREAD

**Issue by FromEndWorld** [allejo/jekyll-toc#25]
I made a demostration post in my blog <https://himring.top/toc-issue/>

When headings go directly from H1 to H3, it seems the TOC cannot recognize them. 

```twig
{% include toc.html html=content %}
```

**Expected TOC**
```html
<li><a href="#h1">H1</a>
  <ul>
    <li>
      <ul>
        <li><a href="#h3-1">H3</a></li>
        <li><a href="#h3-2">H3</a></li>
      </ul>
    </li>
  </ul>
</li>

```

**Comment by allejo** [comment]
Skipping headings causing the TOC to break is a side effect of how this code works but I consider it intended behavior; see my comment in https://github.com/allejo/jekyll-toc/issues/6#issuecomment-358087050. If you *really* do not want to change your headings, there's a patch available in #13 that you can use.

**Comment by FromEndWorld** [comment]
Thanks, the patch is not perfect while it works.
